server/src-lib/Hasura/RQL/DDL/Relationship.hs

```diff
@@ -151,6 +154,51 @@ createObjRelP2 (WithTable qt rd) = do
   objRelP2 qt rd
   return successMsg
 
+runCreateRemoteRelationship ::
+     (MonadTx f, CacheRM f) => CreateRemoteRelationship -> f EncJSON
+runCreateRemoteRelationship createRemoteRelationship = do
+  runCreateRemoteRelationshipP1 createRemoteRelationship
+  runCreateRemoteRelationshipP2 createRemoteRelationship
+
+runCreateRemoteRelationshipP1 ::
+     (MonadTx f, CacheRM f) => CreateRemoteRelationship -> f ()
+runCreateRemoteRelationshipP1 createRemoteRelationship = do
+  sc <- askSchemaCache
+  case HM.lookup
+         (createRemoteRelationshipRemoteSchema createRemoteRelationship)
+         (scRemoteResolvers sc) of
+    Just {} -> do
+      validation <-
+        getCreateRemoteRelationshipValidation createRemoteRelationship
+      case validation of
+        Left err -> throw400 RemoteSchemaError (T.pack (show err))
+        Right {} -> pure ()
+    Nothing -> throw400 RemoteSchemaError "No such remote schema"
+
+runCreateRemoteRelationshipP2 ::
+     (MonadTx f) => CreateRemoteRelationship -> f EncJSON
+runCreateRemoteRelationshipP2 createRemoteRelationship = do
+  liftTx (persistCreateRemoteRelationship createRemoteRelationship)
+  pure successMsg
+
+persistCreateRemoteRelationship
+  :: CreateRemoteRelationship -> Q.TxE QErr ()
+persistCreateRemoteRelationship createRemoteRelationship =
+  Q.unitQE defaultTxErrorHandler [Q.sql|
+  INSERT INTO hdb_catalog.hdb_remote_relationship
+  (name, table_schema, table_name, remote_schema, remote_field, hasura_fields, remote_arguments)
```

Comment:
Should we put the entire configuration into a jsonb column. So the table will only have `name`, `table_schema`, `table_name`,  `remote_schema`, `configuration` ?

I have personally found that configurations are highly volatile and it is just easier to have a json representation of the whole thing as much as possible.

Reply:
If you're happy to spend time on these tweaks, sure. :man_shrugging:

Reply:
Yes, we should. Changing the table afterwards involves writing migration code which can be quite a pain to deal with.

Reply:
Database migrations are a fact of life; if that's a pain to deal with, it's because it wasn't planned for. By storing JSON blobs we're just making the database less structured with less type information, and you'll need to migrate the JSON instances instead, or else throw away user's data if they ever upgrade. But it's not my call. I'm following the project conventions. :man_shrugging:

Reply:
What fields would you like to be in `configuration`? The contents of `CreateRemoteRelationship`?

Reply:
I agree in general. But in this specific case, I am very skeptical of configurations to have fixed type during v1 implementations. 

For e.g. in v2, we are planning to have nested fields joins, remote to remote joins etc. In those cases few of these fields will not even make sense. And then we have a null constraint on these so :man_shrugging:

Reply:
Yeah, lets chuck the whole thing in there.

Reply:
Sure, I'll do that.

> For e.g. in v2, we are planning to have nested fields joins

So in this version, [in these cases](https://github.com/chrisdone/graphql-engine/blob/c77c98948cd1516adbef97969a08adf0855a1ef7/server/src-lib/Hasura/RQL/DDL/Remote/Validate.hs#L177-L179) we could legitimately return an error that non-atomic fields are not permitted?

Reply:
Or else what do you mean by "nested fields joins"?

Reply:
Done the schema change.